homeworks/lesia.turua_lesiaturuta/PokemonProject/scripts/pokemon.js

```diff
@@ -327,28 +106,66 @@ const filterPokemons = (checkBig, checkSmall) => {
     return newPokemons;
 };
 
-const init = () => {
-// disabled for html elements
-    selectSearchEl.disabled = true;
-    selectSearchRadioOnlyNewEl.disabled = true;
-    selectSearchRadioFavoritesEl.disabled = true;
+const filterPokemonsByType = (type) => pokemons.filter((pokemon) => pokemon.type.includes(type));
+
+const init = async () => {
+    // get pokemons
+    try {
+        const url = 'https://my-json-server.typicode.com/electrovladyslav/pokemon-json-server/pokemons';
+        const response = await fetch(url);
+        pokemons = await response.json();
+        filterPokemonsArray = [...pokemons];
+        typesPokemons = updateTypes(pokemons);
+        renderSelect();
+        loadingEl.classList.add('display-none');
+    } catch (e) {
+        alert('Error url, please update this page');
+        console.error(e);
+    }
+
+    // disabled for html elements
+    // selectSearchEl.disabled = true;
+    // selectSearchRadioHighEl.disabled = true;
+    // selectSearchRadioLowEl.disabled = true;
     // selectSearchInputSearchEl.disabled = true;
     // selectSearchInputButtonEl.disabled = true;
 
     selectSearchCheckboxBigEl.addEventListener('change', (e) => {
         checkboxBig = e.target.checked;
         filterPokemonsArray = filterPokemons(checkboxBig, checkboxSmall);
+        typesPokemons = updateTypes(pokemons);
```

Comment:
Couldn't you make one handler for both chackboxes?
Add a proper name, lets say `onSizeFilterChange` and get `checked` state of both checkboxes in it.

Reply:
ok, up to you))